app/controllers/kaui/admin_tenants_controller.rb

```diff
@@ -415,6 +415,55 @@ def catalog_by_effective_date
     end
   end
 
+  def validate_plugin_name
+    json_response do
+      message = 'Official plugin not found'
+      entered_plugin_name = params.require(:plugin_name)
+      nodes_info = KillBillClient::Model::NodesInfo.nodes_info(options_for_klient) || []
+      plugins_info = nodes_info.first.plugins_info || []
+      splitted_entered_plugin_name = split_camel_dash_underscore_space(entered_plugin_name)
+
+      weights = []
+      found_plugin = nil
+      plugins_info.each do |plugin|
+        next if plugin.version.nil?
+        if plugin.plugin_name == entered_plugin_name || plugin.plugin_key == entered_plugin_name
+          message = nil
+          weights = []
+          found_plugin = plugin
+          break
+        end
+
+        splitted_plugin_name = split_camel_dash_underscore_space(plugin.plugin_name)
+        weight = { :plugin_name => plugin.plugin_name, :plugin_key => plugin.plugin_key, :worth_weight => 0 }
+        splitted_entered_plugin_name.each do |entered|
+          if splitted_plugin_name.include?(entered)
+            weight[:worth_weight] = weight[:worth_weight] + 1
+          end
+
+          splitted_plugin_name.each do |splitted|
+            if entered.chars.all? { |ch| splitted.include?(ch) }
+              weight[:worth_weight] = weight[:worth_weight] + 0.5
+              break
+            end
+          end
+        end
+        weights << weight if weight[:worth_weight] > 0
+
+      end
+
+      if weights.size > 0
+        weights.sort! { |a,b| b[:worth_weight] <=> a[:worth_weight] } if weights.size > 1
+        plugin_anchor = view_context.link_to(weights[0][:plugin_name], '#', id: 'suggested',
```

Comment:
Maybe I am just confused by the code, but isn't `weights[0]` at this point the best match found running on the server? If so, the message should warn the user that a similar plugin is already installed. The ticket didn't have a suggestion for the message, but maybe we can say something along the lines of "Similar plugin already installed: #{plugin_anchor}'".


For the _did you mean_, the source of truth is https://github.com/killbill/killbill-cloud/blob/master/kpm/lib/kpm/plugins_directory.yml, see https://github.com/killbill/killbill-admin-ui/blob/3a9650fa85167eedb30a8e70d83465255a849dfe/app/models/kaui/admin_tenant.rb#L29-L38